src/routes/jira/configuration/jira-configuration-get.ts

```diff
@@ -94,30 +118,41 @@ export const JiraConfigurationGet = async (
 		req.log.debug("Received jira configuration page request");
 
 		const subscriptions = await Subscription.getAllForHost(jiraHost);
-		const installations = await getInstallations(subscriptions, req.log, gitHubAppId);
-
-		const failedConnections: FailedConnection[] = await Promise.all(
-			installations.rejected.map(async (installation) => {
-				const sub = subscriptions.find((sub: Subscription) => installation.id === sub.gitHubInstallationId);
-				const repo = sub && await RepoSyncState.findOneFromSubscription(sub);
-				return {
-					id: installation.id,
-					deleted: installation.deleted,
-					orgName: repo?.repoOwner
-				};
-			}));
-
-		const successfulConnections: SuccessfulConnection[] = installations.fulfilled
-			.map((installation) => ({
-				...installation,
-				isGlobalInstall: installation.repository_selection === "all"
-			}));
-
-		const handleNavigationClassName = await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost)
-			? "select-github-version-link"
-			: "add-organization-link";
-
-		res.render("jira-configuration.hbs", {
+		const { installations, successfulConnections, failedConnections } = await getConnectionsAndInstallations(subscriptions, req, gitHubAppId);
+
+		const gheServers = await GitHubServerApp.getForInstallationId(res.locals.installation.id);
+
+		// Grouping the list of servers by `gitHubBaseUrl`
+		const modifiedGHEServers = await Promise.all(chain(gheServers).groupBy("gitHubBaseUrl")
+			.map(async (value, key) => ({
+				gitHubBaseUrl: key,
+				// Fetching the subscriptions for each GH Server App
+				applications: await Promise.all(value.map(async (app) => {
+					const {
+						successfulConnections,
+						failedConnections
+					} = await getConnectionsAndInstallations(subscriptions, req, app.id);
+					app["successfulConnections"] = successfulConnections;
+					app["failedConnections"] = failedConnections;
+
+					return app;
+				}))
+			})).value());
+
+
+		const gheServerEnabled = await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost);
```

Comment:
Don't you want to evaluate this FF before doing all the work above?  One of the reason to have an FF is that if something goes wrong in production, you can turn it off and it'll work like it did previously, but you've changed so much logic that if something in that code above doesn't work, it'll break with no way to turn it off other than reverting the PR and redeploying.

Reply:
The logic for this method has been updated by just one if-else, if FF is true, fetch config from `JiraCloudAndEnterpriseConfiguration`, else fetch from `JiraCloudConfiguration`.